source/grid/tria.cc

```diff
@@ -12520,6 +12520,34 @@ void Triangulation<dim, spacedim>::copy_triangulation(
   if (other_tria.policy)
     this->policy = other_tria.policy->clone();
 
+  // periodic faces
+  this->periodic_face_pairs_level_0.reserve(
+    other_tria.periodic_face_pairs_level_0.size());
+
+  for (auto entry : other_tria.periodic_face_pairs_level_0)
+    {
+      entry.cell[0] =
+        cell_iterator(this, entry.cell[0]->level(), entry.cell[0]->index());
+      entry.cell[1] =
+        cell_iterator(this, entry.cell[1]->level(), entry.cell[1]->index());
+      periodic_face_pairs_level_0.emplace_back(entry);
```

Comment:
I find this very difficult to read because from the `for (auto entry : other_entry...)`, I had assumed that `entry` corresponds to the entries of the *other* triangulation. In fact, your code relies on the fact that `other_tria.periodic_face_pairs_level_0` uses iterators that are pointers to objects, rather than reference objects.

I think this would be substantially easier to read if you wrote it as follows:
```suggestion
  for (const auto &other_entry : other_tria.periodic_face_pairs_level_0)
    {
       auto entry = other_entry;
      entry.cell[0] =
        cell_iterator(this, entry.cell[0]->level(), entry.cell[0]->index());
      entry.cell[1] =
        cell_iterator(this, entry.cell[1]->level(), entry.cell[1]->index());
      periodic_face_pairs_level_0.emplace_back(entry);
```

Reply:
That is true. Thank you for reviewing, I have updated the loop now.